compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -93,6 +94,56 @@ pub(super) fn search_for_section<'a>(
         .map_err(|e| format!("failed to read {} section in '{}': {}", section, path.display(), e))
 }
 
+fn add_gnu_property_note(
+    sess: &Session,
+    file: &mut write::Object<'static>,
+    architecture: Architecture,
+    binary_format: BinaryFormat,
+) {
+    // check bti protection
+    let mut check_cfprotection = false;
+    let mut pr_type: u32 = 0;
+    if architecture == Architecture::X86_64
+        && matches!(
+            sess.opts.unstable_opts.cf_protection,
+            rustc_session::config::CFProtection::Branch | rustc_session::config::CFProtection::Full
+        )
+    {
+        check_cfprotection = true;
+        pr_type = 0xc0000002;
+    }
+    if architecture == Architecture::Aarch64 && sess.opts.unstable_opts.branch_protection.is_some()
+    {
+        check_cfprotection = true;
+        pr_type = 0xc0000000;
```

Comment:
on LLVM side we use these defines, maybe worth to add them to object::elf.
https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/BinaryFormat/ELF.h#L1660

Reply:
I agree. I created a PR for that crate: https://github.com/gimli-rs/object/pull/537